### PR TITLE
test: move oldtests to test directory

### PR DESCRIPTION
**Pull request by dundargoc**
The new oldtest directory is in test/old/testdir. The reason for this is
that many tests have hardcoded the parent directory name to be
'testdir'.

### CODE REVIEW

**Comment by dundargoc** [comment]
See also: https://github.com/neovim/neovim/issues/21559#issuecomment-1368248698

**Review by zeertzjq — DISMISSED** [review]
Also update `vim-patch.sh`

**Comment by zeertzjq** [comment]
Oops, can't dismiss review.

File: .gitattributes [hunk 1]
```diff
@@ -8,6 +8,6 @@ src/xdiff/**    linguist-vendored
 src/cjson/**    linguist-vendored
 src/unicode/**  linguist-vendored
 
-src/nvim/testdir/test42.in diff
+test/testdir/test42.in diff
```
Comment (zeertzjq) [inline .gitattributes:12]: This file doesn't exist
```suggestion
```

**Comment by zeertzjq** [comment]
https://github.com/vim/vim/pull/12111

**Review by zeertzjq — APPROVED** [review]

**Review by justinmk — APPROVED** [review]
Nice!

**Pull request merged**